### PR TITLE
Fix react warnings

### LINKED ISSUE #715: Fail tests if React throws warnings
When running tests I notice there are a lot of warnings related to duplicate keys and failed prop types. I don't think these warnings should be ignored.

**Describe the solution you'd like**
- We should fail tests when React outputs warnings
- This issue should involve creating a PoC PR to fail the tests in this way
- It should also involve summarising the number and type of warnings, estimate effort involved to fix them  and propose a plan for breaking down the work required.

**Testing notes**
I think the chance of a regression is low and more likely to fix any problems that are currently uncaught.

**Related issues**
- https://github.com/bbc/simorgh/issues/289 describes making prop type warnings fails and the challenges of doing this

### PR DESCRIPTION
**Pull request by jroebu14**
Resolves https://github.com/bbc/psammead/issues/715

**Overall change:** Fixes some of React warnings in our tests. There are at least 26 warnings remaining to be fixed.

**Code changes:**

- Fixes warnings for various components
  - footer
  - paragraph
  - default page wrapper
  - media indicator
  - link contents
  - metadata
  - front page
  - audio video head
  - image with placeholder
  - caption
  - article
  - error page
  - icon links
  - storypromo

- Fails a unit test if the suite contains more than 3 React warnings. Ideally there should be zero React warnings per test suite however fixing all in one go is too big a task in Simorgh. We need to keep fixing and lowering the warnings limit until all tests have no warnings.
---

- [x] I have assigned myself to this PR and the corresponding issues
- [ ] Tests added for new features
- [ ] Test engineer approval

**Testing notes**

Mostly test file changes but some very light refactoring of page metadata - `src/app/components/Metadata/index.jsx`

### CODE REVIEW

**Comment by jroebu14** [comment]
The remaining React warnings in Simorgh has been broken down into separate issues:

bbc/simorgh#2784
bbc/simorgh#2783
bbc/simorgh#2782
bbc/simorgh#2777
bbc/simorgh#2776
bbc/simorgh#2775
bbc/simorgh#2774
bbc/simorgh#2773
bbc/simorgh#2771
bbc/simorgh#2770
bbc/simorgh#2769
bbc/simorgh#2768
bbc/simorgh#2767
bbc/simorgh#2766
bbc/simorgh#2765
bbc/simorgh#2764
bbc/simorgh#2763
bbc/simorgh#2762
bbc/simorgh#2760

File: src/app/models/propTypes/storyItem/index.js [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import { shape, string, number } from 'prop-types';
 
 export const storyItemImage = {
-  path: string.isRequired,
-  altText: string.isRequired,
-  height: number.isRequired,
-  width: number.isRequired,
+  path: string,
```
Comment (chris-hinds) [inline src/app/models/propTypes/storyItem/index.js:4]: should these not be required?

Comment (jroebu14) [reply src/app/models/propTypes/storyItem/index.js:4]: @hindsc52 if you look at a [component using these props](https://github.com/bbc/simorgh/blob/11d855d1ecfb7065af3d113598a9fcf93af05a27/src/app/containers/StoryPromo/index.jsx#L76-L99) you can see they are actually optional

**Review by chris-hinds — APPROVED** [review]

File: src/app/models/propTypes/storyItem/index.js (hunk 1 above)
Comment (jroebu14) [reply src/app/models/propTypes/storyItem/index.js:4]: also here https://github.com/bbc/simorgh/blob/11d855d1ecfb7065af3d113598a9fcf93af05a27/src/app/containers/StoryPromo/index.jsx#L22

**Review by dr3 — APPROVED** [review]

**Comment by PriyaKR** [comment]
@jroebu14 i dont see the unit tests failing when there has been more than three warnings.

**Comment by jroebu14** [comment]
@PriyaKR it will be 3 warnings per test file. Unfortunately it's not possible to make the unit tests fail on 3 warnings for all test suites.

**Comment by PriyaKR** [comment]
I can see tests passing when there has been four warnings in a test file.
unit_test_react_warnings

**Comment by jroebu14** [comment]
@PriyaKR I've figured out why the unit tests weren't failing. However there are now up to 6 failing tests in a test file which means I either need to do quite a bit more test fixing to limit warnings to 3 or increase the warnings limit to 6. As I need to continue with media work I think the latter of the 2 is the best option for me. It's not really a great solution to allow up to 6 warnings per test file though 😞

**Comment by PriyaKR** [comment]
LGTM.

**Pull request merged**